instrumentation/rdkafka/lib/opentelemetry/instrumentation/rdkafka/patches/consumer.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Rdkafka
+      module Patches
+        # The Consumer module contains the instrumentation patch for the Consumer class
+        module Consumer
+          def each
+            super do |message|
+              attributes = {
+                'messaging.system' => 'kafka',
+                'messaging.destination' => message.topic,
+                'messaging.destination_kind' => 'topic',
+                'messaging.kafka.partition' => message.partition,
+                'messaging.kafka.offset' => message.offset
+              }
+
+              attributes['messaging.kafka.message_key'] = message.key if message.key
```

Comment:
This PR may be relevant: https://github.com/open-telemetry/opentelemetry-ruby/pull/1059

How does `rdkafka` deal with binary keys?

Reply:
After some testing with some binary keys, rdkafka appears to automatically handle key decoding.